docs/clustering.md

```diff
@@ -1,92 +1,168 @@
+AWX supports multi-node configurations. Here is an example configuration with two control plane nodes.
 
-* Each instance should be able to act as an entry point for UI and API Access.
-  This should enable AWX administrators to use load balancers in front of as many instances as they wish and maintain good data visibility.
-* Each instance should be able to join the AWX cluster and expand its ability to execute jobs.
-* Provisioning new instance should be as simple as updating the `inventory` file and re-running the setup playbook.
-* Instances can be de-provisioned with a simple management command.
-* Instances can be grouped into one or more Instance Groups to share resources for topical purposes.
-* These instance groups should be assignable to certain resources:
-  * Organizations
-  * Inventories
-  * Job Templates
+```
+       ┌───────────────────────────┐
+       │      Load-balancer        │
+       │   (configured separately) │
+       └───┬───────────────────┬───┘
+           │   round robin API │
+           ▼       requests    ▼
+
+  AWX VM/Pod 1               AWX VM/Pod 2
+┌──────────────┐           ┌──────────────┐
+│              │           │              │
+│ ┌──────────┐ │           │ ┌──────────┐ │
+│ │ awx-task │ │           │ │ awx-task │ │
+│ ├──────────┤ │           │ ├──────────┤ │
+│ │ awx-ee   │ │           │ │ awx-ee   │ │
+│ ├──────────┤ │           │ ├──────────┤ │
+│ │ awx-web  │ │           │ │ awx-web  │ │
+│ ├──────────┤ │           │ ├──────────┤ │
+│ │ redis    │ │           │ │ redis    │ │
+│ └──────────┘ │           │ └──────────┘ │
+│              │           │              │
+└──────────────┴─────┬─────┴──────────────┘
+                     │
+                     │
+               ┌─────▼─────┐
+               │ Postgres  │
+               │ database  │
+               └───────────┘
+```
+
+There are two main deployment types, virtual machines (VM) or K8S. Ansible Automation Controller installation supports both VM or K8S deployments. The upstream AWX project only supports K8S deployments. Either deployment type supports cluster scaling.
+- Control plane nodes run a number of background services that are managed by supervisord
+  - dispatcher
+  - wsbroadcast
+  - callback receiver
+  - receptor (*managed under systemd)
+  - redis (*managed under systemd)
+  - uwsgi
+  - daphne
+  - rsyslog
+- For K8S deployments, these background processes are containerized
+  - `awx-ee`: receptor
+  - `awx-web`: uwsgi, daphne, wsbroadcast, rsyslog
+  - `awx-task`: dispatcher, callback receiver
+  - `redis`: redis
+- Each control node is monolithic and contains all the necessary components for handling API requests and running jobs.
```

Comment:
for now... will change soon. But maybe this is a good strategy, as your updated wording makes it very clear where the new web/task split feature needs to update.